### ISSUE TITLE
Feature Request: Add a new section in Model Zoo for Generation

### ISSUE THREAD

**Issue by rahulsnkr** [Cloud-CV/Fabrik#489]
Generative Adversarial Networks (GANs) have gained a lot of traction, both from research point of view, as well as development, due to their numerous applications. A lot of GAN architectures have also been published, for example - 
* [Vanilla GAN](https://arxiv.org/abs/1406.2661)
* [DCGAN](https://arxiv.org/abs/1511.06434)
* [Conditional GAN](https://arxiv.org/abs/1411.1784)

Therefore, it makes sense to have a separate section for Generation in the Model Zoo, and GAN architectures could be added within that section

I'd like to work on this, starting with Vanilla GAN

**Comment by Ram81** [comment]
@rahulsnkr sure go ahead

**Comment by rahulsnkr** [comment]
At the moment, Fabrik is not capable of importing/exporting two networks at the same time, and the given issue is dependent on the same, since a GAN consists of two networks, i.e. a generator and a discriminator. How should I proceed from here? Should I try to add support for Fabrik so that it is capable of supporting more than one network?

**Comment by thatbrguy** [comment]
@rahulsnkr In my opinion supporting more than one network does not sound like a good idea. Other than massive UI redesign, we also have to take care of proper usage of graphs and sessions in Keras and TF. 

Instead of that, we can just use Generators and Discriminators separately. To be honest, most Generators are variants of U-Nets and Autoencoders. Similarly, Discriminators are straightforward too, except for some like the PatchGAN. Keeping Generators and Discriminators separate would also allow you to mix and match combinations.

What you can do instead is while exporting a Generator we could have a popup that allows you to download a Discriminator as well. We can also have toy code samples to enable loading a GAN and using them. @Ram81 What do you think?

**Comment by rahulsnkr** [comment]
@thatbrguy yes that seems like a much better idea. It will also give the users more flexibility. Maybe the 'Generation' section can have two subsections for generators and discriminators, and as you said, the user can be prompted to download a discriminator when exporting a generator, and vice versa. @Ram81 does this sound ok?

**Comment by kartikdutt18** [comment]
@rahulsnkr @Ram81 If generators and discriminators are made available separately, Can I work on DCGAN ?

**Comment by Ram81** [comment]
@rahulsnkr can you research if it's feasible to have multiple model in fabrik's current implementation. If it is not feasible then we can go ahead with some other approach

**Comment by rahulsnkr** [comment]
@Ram81 i'll look into it!

**Comment by rohanasthana** [comment]
Is this still open? I would like to work on this issue
